message.go

```diff
@@ -384,30 +384,29 @@ type Arena interface {
 	Allocate(minsz Size, segs map[SegmentID]*Segment) (SegmentID, []byte, error)
 }
 
-type singleSegmentArena []byte
+// SingleSegmentArena is a simple arena comprising an expanding buffer.
+type SingleSegmentArena []byte
 
 // SingleSegment returns a new arena with an expanding single-segment
 // buffer.  b can be used to populate the segment for reading or to
 // reserve memory of a specific size.  A SingleSegment arena does not
 // return errors unless you attempt to access another segment.
 func SingleSegment(b []byte) Arena {
```

Comment:
You probably want to change the return type of this function to match, yeah?

Reply:
Fixed